packages/shared/useDateFormat/index.ts

```diff
@@ -0,0 +1,72 @@
+import type { MaybeRef } from '@vueuse/shared'
+import { computed, unref } from 'vue-demi'
+
+export type UDate = Date | number | string | undefined
+
+const REGEX_PARSE = /* #__PURE__ */ /^(\d{4})[-/]?(\d{1,2})?[-/]?(\d{0,2})[Tt\s]*(\d{1,2})?:?(\d{1,2})?:?(\d{1,2})?[.:]?(\d+)?$/
+const REGEX_FORMAT = /* #__PURE__ */ /\[([^\]]+)]|Y{1,4}|M{1,4}|D{1,2}|d{1,4}|H{1,2}|h{1,2}|a|A|m{1,2}|s{1,2}|Z{1,2}|SSS/g
+
+export const padStart = (string: number, length: number, pad: string) => {
```

Comment:
Can't we just use padStart from the string prototype? If not, we'd better to move this to utils

Reply:
> Can't we just use padStart from the string prototype? If not, we'd better to move this to utils

I'm not so sure about that, because it may have poor compatibility, do we need to consider it ?

Reply:
> > Can't we just use padStart from the string prototype? If not, we'd better to move this to utils
> 
> I'm not so sure about that, because it may have poor compatibility, do we need to consider it ?

ES2017 include `String.prototype.padStart`
https://github.com/vueuse/vueuse/blob/main/tsconfig.json#L3